### PR TITLE
fixed-region-provider-data-is-unavailable

### PR DESCRIPTION
**Pull request by vchikoti1998**
region-provider-data-missing-default-data
## Types of changes

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
region provider data missing default data.

## Related Issue(s)
https://github.com/aws/aws-toolkit-jetbrains/issues/3264
## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if change is customer facing in the IDE.
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: core/src/software/aws/toolkits/core/parser/JsonUtilParser.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.core.parser
+
+import software.aws.toolkits.core.region.PartitionParser
+import java.io.InputStream
+
+class JsonUtilParser : Parser() {
+    override fun isParseSuccess(data: InputStream): Boolean {
+        return PartitionParser.parse(data)?.partitions?.isEmpty() ?: return false
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply core/src/software/aws/toolkits/core/parser/JsonUtilParser.kt]: This is what agreed with Richard.

File: core/src/software/aws/toolkits/core/parser/XmlParser.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import javax.xml.stream.XMLInputFactory
+import javax.xml.stream.XMLStreamException
+import javax.xml.stream.XMLStreamReader
+
+class XmlParser : Parser() {
+    override fun isParseSuccess(data: InputStream): Boolean {
+        val factory: XMLInputFactory = XMLInputFactory.newFactory()
+        factory.setProperty(XMLInputFactory.SUPPORT_DTD, false)
+        factory.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES, false)
+        val reader: XMLStreamReader = factory.createXMLStreamReader(data)
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply core/src/software/aws/toolkits/core/parser/XmlParser.kt:16]: specific object is required for parsing, when we use xmlMapper.

File: core/tst/software/aws/toolkits/core/utils/RemoteResourceResolverTest.kt [hunk 3]
```diff
@@ -44,15 +44,15 @@ class RemoteResourceResolverTest {
         val secondCall = sut.resolve(resource).unwrap()
 
         assertThat(firstCall).isEqualTo(secondCall)
-        assertThat(firstCall).hasContent("data")
+        assertThat(firstCall).hasContent(DATA)
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply core/tst/software/aws/toolkits/core/utils/RemoteResourceResolverTest.kt:46]: Right now, we are handling only JSON and XML files and confirmed with Richard about this.

File: core/src/software/aws/toolkits/core/parser/Parser.kt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.core.parser
+
+import java.io.InputStream
+
+abstract class Parser {
+    abstract fun isParseSuccess(data: InputStream): Boolean
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply core/src/software/aws/toolkits/core/parser/Parser.kt:9]: Its Done.

File: core/src/software/aws/toolkits/core/parser/Parser.kt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.core.parser
+
+import java.io.InputStream
+
+abstract class Parser {
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply core/src/software/aws/toolkits/core/parser/Parser.kt:8]: Its Done.

[the remark this answers is not in the document]
Comment (vchikoti1998) [reply core/src/software/aws/toolkits/core/parser/Parser.kt:8]: Should I move Parser Interface into RemoteResourceResolverProvider.kt ?

File: core/src/software/aws/toolkits/core/lambda/LambdaSampleEventProvider.kt [hunk 6]
```diff
@@ -64,4 +65,5 @@ internal data class LambdaSampleEventResource(val filename: String) : RemoteReso
 
     override val name: String = "lambda-sample-event-$filename"
     override val ttl: Duration = Duration.ofDays(7)
+    override val parser: XmlParser = XmlParser()
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply core/src/software/aws/toolkits/core/lambda/LambdaSampleEventProvider.kt:68]: I didn't follow, can you explain more.

File: core/src/software/aws/toolkits/core/parser/XmlParser.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply core/src/software/aws/toolkits/core/parser/XmlParser.kt:16]: I didn't realize that we have in LambdaSampleEventProvider.kt, i will update it.

File: core/tst/software/aws/toolkits/core/utils/RemoteResourceResolverTest.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply core/tst/software/aws/toolkits/core/utils/RemoteResourceResolverTest.kt:46]: Like how we are validating XML and JSON, Should the plaintext file in RemoteResourceResolver.kt be validated as well?

File: core/src/software/aws/toolkits/core/parser/Parser.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply core/src/software/aws/toolkits/core/parser/Parser.kt:9]: What is the need for passing parsing result here? all we need is parsing success or not right.

File: core/tst/software/aws/toolkits/core/utils/RemoteResourceResolverTest.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (rli) [reply core/tst/software/aws/toolkits/core/utils/RemoteResourceResolverTest.kt:46]: we should support the idea of "no validation" in the case where we don't have a good heuristic

File: core/src/software/aws/toolkits/core/lambda/LambdaSampleEventProvider.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (rli) [reply core/src/software/aws/toolkits/core/lambda/LambdaSampleEventProvider.kt:68]: the manifest is XML, but the events themselves are JSON

**Pull request closed without merging**